CHANGELOG.md

```diff
@@ -16,6 +16,7 @@
 * [CHANGE] Added new metric `cortex_compactor_disk_out_of_space_errors_total` which counts how many times a compaction failed due to the compactor being out of disk. #8237
 * [CHANGE] Anonymous usage statistics tracking: report active series in addition to in-memory series. #8279
 * [CHANGE] Ruler: `evaluation_delay` field in the rule group configuration has been deprecated. Please use `query_offset` instead (it has the same exact meaning and behaviour). #8295
+* [CHANGE] Distributor: `distributor.Push()` errors with error cause `mimirpb.BAD_DATA` are now mapped to gRPC status code `codes.InvalidArgument` instead of `codes.FailedPrecondition`.
```

Comment:
When we introduced gRPC codes on distributor side, we documented that change in CHANGELOG and explained which (ex)HTTP status now gets mapped to which gRPC status ([here](https://github.com/grafana/mimir/blob/51c4088cc3b5087aabf72478dcc909d011ff64c0/CHANGELOG.md?plain=1#L335)). I thought that now it makes sense to write the update, otherwise the last documented status remains the one that is valid. WDYT @pstibrany?

Reply:
But is this visible to user in any way? Because if not, why does user care? Perhaps we should remove the previous changelog entry that you refer to?

Reply:
Well, we export the `distributor.Distributor` struct, which is an implementation of `distributorpb.DistributorServer` interface. IMO it is important that the exported `Distributor.Push()` changed its semantics, and that in case of a validation error it returns a gRPC error with code `InvalidArgument` instead of `FailedPrecondition`. This will also be shown in the Distributor's request duration metrics when the `/distributor.Distributor/Push` endpoint is called.

But if you prefer that we remove both CHANGELOG entries I can do that.

Reply:
External users are not supposed to use Mimir as a library, or use `distributor.Distributor` struct.

However we embed Mimir in our enterprise product, and you should check if we use gRPC codes there, and update it accordingly. (I believe our Adaptive Metrics code does that)

To sum it up: please remove the changelog entry, and please integrate this change into our enterprise product, after this PR is merged.

Reply:
I have removed the CHANGELOG entries. 
I will also check the adaptive metrics code. It should be already OK because I fixed it during the error handling improvement work, but I will double check once this PR is merged.
Thank you.

Reply:
Thank you.